### PR TITLE
feat: Add option to disable lambda to sync runner binaries

### PR DESCRIPTION
**Pull request by npalm**
Create option to disable the deployment of the binary syncer module, typically not needed when having pre-build AMI containing the GitHub runner.

## What is changed?
- Added variable `enable_runner_binaries_syncer` to disable syncer
- Refactored the internals of the runner moduel. When using the internal modules directoy be-aware of breaking changes, you have to set `s3_runner_binaries`.
- Updated example for pre-build AMI to disable the syncer module.
- Narrowed down the bucket permission of the runner to only the relevant distribution.

## Testes
- Default example
- Pre-build example

### CODE REVIEW

**Comment by npalm** [comment]
Close #2274

**Review by GuptaNavdeep1983 — DISMISSED** [review]

File: main.tf [hunk 1]
```diff
@@ -134,8 +133,14 @@ module "runners" {
   prefix        = var.prefix
   tags          = local.tags
 
-  s3_bucket_runner_binaries   = module.runner_binaries.bucket
-  s3_location_runner_binaries = local.s3_action_runner_url
+  # s3_bucket_runner_binaries   = var.enable_runner_binaries_syncer ? module.runner_binaries[0].bucket : null
```
Comment (GuptaNavdeep1983) [inline main.tf:136]: Should have this commented code?

File: modules/runners/variables.tf [hunk 2]
```diff
@@ -566,4 +564,10 @@ variable "lambda_architecture" {
...
     error_message = "`lambda_architecture` value is not valid, valid values are: `arm64` and `x86_64`."
   }
-}
\ No newline at end of file
+}
+
+variable "enable_runner_binaries_syncer" {
+  description = "Option to disable the lambda to sync GitHub runner distribution, usefull when using a pre-build AMI."
+  type        = bool
+  default     = true
```
Comment (GuptaNavdeep1983) [inline modules/runners/variables.tf:572]: Do we want to enable it by default?

Comment (npalm) [reply modules/runners/variables.tf:572]: Yes since it is current enabled by design and this way we don't force migration from current users.

**Review by GuptaNavdeep1983 — APPROVED** [review]

**Pull request merged**